src/engine/map/grid.ts

```diff
@@ -182,8 +177,18 @@ export class Grid {
     const isClaimableRoute = track.isClaimable() || track.wasClaimed();
     return tupleMap(track.getExits(), (exit) => {
       const otherExit = track.otherExit(exit);
-      const connects =
-        exit === TOWN || this.connection(track.coordinates, exit) != null;
+      let connects: boolean;
```

Comment:
[Side-note] I tend to hate let statements as I find them difficult to reason about, and I have found that attempting to eliminate them makes for better code.

For example, you could adjust this series of if statements into a series of individual booleans, then build connects from that.

```
const isTown = ...
const neighborIsCity = ...
const hasTrackConnection = ...
const connects = isTown || neighborIsCity || hasTrackConnection;
```

Reply:
Nice, I do like pattern more. Updating this.